pkg/alertmanager/multitenant.go

```diff
@@ -452,7 +460,11 @@ func (am *MultitenantAlertmanager) ServeHTTP(w http.ResponseWriter, req *http.Re
 	am.alertmanagersMtx.Lock()
 	userAM, ok := am.alertmanagers[userID]
 	am.alertmanagersMtx.Unlock()
-	if !ok {
+
+	userAM.activeMtx.Lock()
```

Comment:
Have you considered moving this logic (starting from this line until the end of the function) to a `ServeHTTP()` function in the alertmanager? From a design perspective, I think it would be better if we keep the `activeMtx` usage isolated within the alertmanager.

Reply:
I think it's fair to have check for active and return early here, but accessing Active flag definitely does belong to alertmanager, to keep locking in one place only. Also `mux` should not be accessed directly.

```go
    if !ok || !userAM.isActive() { // isActive should do the locking
        ...
    }
    userAM.ServeHTTP(w, req)
```

Reply:
👍